lib/modules/manager/tekton/extract.ts

```diff
@@ -0,0 +1,100 @@
+import is from '@sindresorhus/is';
+import { loadAll } from 'js-yaml';
+import { logger } from '../../../logger';
+import { getDep } from '../dockerfile/extract';
+import type { PackageDependency, PackageFile } from '../types';
+import type { TektonBundle, TektonResource } from './types';
+
+export function extractPackageFile(
+  content: string,
+  fileName: string
+): PackageFile | null {
+  logger.trace('tekton.extractPackageFile()');
+  const deps: PackageDependency[] = [];
+  let docs: TektonResource[];
+  try {
+    docs = loadAll(content) as TektonResource[];
+  } catch (err) {
+    logger.debug(
+      { err, fileName },
+      'Failed to parse YAML resource as a Tekton resource'
+    );
+    return null;
+  }
+  for (const doc of docs) {
+    deps.push(...getDeps(doc));
+  }
+  if (!deps.length) {
+    return null;
+  }
+  return { deps };
+}
+
+function getDeps(doc: TektonResource): PackageDependency[] {
+  const deps: PackageDependency[] = [];
+  if (is.falsy(doc)) {
+    return deps;
+  }
+
+  // Handle TaskRun resource
+  addDep(doc.spec?.taskRef, deps);
+
+  // Handle PipelineRun resource
+  addDep(doc.spec?.pipelineRef, deps);
+
+  // Handle Pipeline resource
+  for (const task of doc.spec?.tasks ?? []) {
+    addDep(task.taskRef, deps);
+  }
+
+  // Handle TriggerTemplate resource
+  for (const resource of doc.spec?.resourcetemplates ?? []) {
+    addDep(resource?.spec?.taskRef, deps);
+    addDep(resource?.spec?.pipelineRef, deps);
+  }
+
+  // Handle list of TektonResources
+  const items = doc.items ?? [];
+  if (items.length > 0) {
+    for (const item of items) {
+      deps.push(...getDeps(item));
+    }
+  }
+
+  return deps;
+}
+
+function addDep(ref: TektonBundle, deps: PackageDependency[]): void {
+  if (is.falsy(ref)) {
+    return;
+  }
+  let imageRef = '';
+  // Find a bundle reference from the Bundle resolver
+  if (ref.resolver === 'bundles') {
+    for (const field of ref.resource ?? []) {
+      if (field.name === 'bundle') {
+        imageRef = field.value;
+        break;
+      }
+    }
+  }
+  if (!is.string(imageRef) || is.emptyString(imageRef)) {
+    // Fallback to older style bundle reference
+    imageRef = ref.bundle;
+    if (!is.string(imageRef) || is.emptyString(imageRef)) {
```

Comment:
there is a `is.notEmptyAndNotWhitespaceString` function to combine both conditions. 😉

Reply:
I don't see that one. I see `nonEmptyStringAndNotWhitespace` and `emptyStringOrWhitespace` which are slightly different and don't guard against the case where `imageRef` is not a string at all which I think is something important to guard against.

I'm surprised that imageRef in some cases could be something other than a string given that the types have been defined. This sort of seems like an odd behavior in TypeScript. For this document:

```
kind: Pipeline    
spec:             
  tasks:          
  - taskRef:      
      bundle: true
```

`typeof imageRef` returns `'boolean'`. However, `is.boolean(imageRef)` returns `false`. I'm digging more into this.

Reply:
Hmm it seems that this is a limitation where the type enforcement is not done at runtime. I'll change the code to assume it's a string, and let the dependency have a `skipReason: invalid-value`. This is likely the most correct way to approach this as at that point, there's an invalid Tekton resource defined.